### PR TITLE
WIP: Repository restructure

### PR DESCRIPTION
**Pull request by FinnStutzenstein**
- moved all server related things into the folder `server`, so this
configuration is parallel to the client.
- All main "services" are now folders in the root directory
- Added Dockerfiles to each service (currently server and client)
- Added a docker compose configuration to start everything together.
Currently there are heavy dependencies into https://github.com/OpenSlides/openslides-docker-compose
- resturctured the .gitignore. If someone needs something excluded,
please add it to the right section.

Startup of openslides with docker compose:
- Navigate into `docker`
- `./build.sh`
- `docker-compose up` (Add a `-d` if you want to dispatch it)
- You can start and stop the configuration with
`docker-compose start`/`docker-compose stop`. This will reuse all
existing containers (and volumes, ...)
- Cleanup: `docker-compose down` will remove all containers

Important things to consider:
- Please make use of `docker/secrets/` to autocreate initial users
- The dockerfile does not contain any mounted volumes. Make sure to safe
your data!
- postfix might not be configured right

@gsiv We should review this together.

### CODE REVIEW

**Comment by FinnStutzenstein** [comment]
TODOS:
- [ ] travis
- [ ] postfix config
- [ ] amount default pgnodes
- [ ] upstream port argument for the `docker-compose.yml`
- [ ] adding a `docker-stack.yml`?

**Comment by tsiegleauq** [comment]
building, adding secrets and running docker worked fine for me.

The folder structure is better than before (imo), I only feel like the naming of the "build.sh" scripts are a little misleading.
You'd instinctively guess that the build script in client would prod-build the angular application and not just the docker container.
If the tool chain is not strictly dependent to a "build.sh"-file, please rename them to "build-docker-container.sh"

**Comment by FinnStutzenstein** [comment]
Thanks for the feedback!

@gsiv will do some more tooling and replaceing these temporary scripts.

File: DEVELOPMENT.rst [hunk 1]
```diff
@@ -37,15 +37,30 @@ Clone current master version from `OpenSlides GitHub repository
 c. Setup a virtual Python environment (optional)
 ''''''''''''''''''''''''''''''''''''''''''''''''
 
-See step 1. b. in the installation section in the `README.rst
-<https://github.com/OpenSlides/OpenSlides/blob/master/README.rst>`_.
+You can setup a virtual Python environment using the virtual environment
+(venv) package for Python to install OpenSlides as non-root user. This will
+allow for encapsulated dependencies. They will be isntalled in the virtual
```
Comment (FinnStutzenstein) [inline DEVELOPMENT.rst:42]: typo

File: server/docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+# Variables relevant for CMD
+ENV DJANGO_SETTINGS_MODULE settings
+ENV PYTHONPATH personal_data/var/
+
+RUN mkdir -p /app
+WORKDIR /app
+RUN useradd -m openslides
+RUN chown -R openslides /app
+
+RUN apt-get -y update && apt-get -y upgrade && \
```
Comment (FinnStutzenstein) [inline server/docker/Dockerfile:17]: @gsiv Please remove tools, that are not required anymore.

**Comment by FinnStutzenstein** [comment]
@gsiv note: I rebased on the current master

**Comment by tsiegleauq** [comment]
>19 commits
was this necessary?

**Pull request merged**